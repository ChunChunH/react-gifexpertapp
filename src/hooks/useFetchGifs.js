import {useState, useEffect} from 'react';
import getGifs from '../helpers/getGifs';


const useFetchGifs = (category) => {


    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
   
        getGifs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            })

    }, [category] /*si lo pones vacio lo ejecuta 1 vez*/)



    return state;

}

export default useFetchGifs



