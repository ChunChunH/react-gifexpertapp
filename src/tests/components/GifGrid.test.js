import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import React from "react"
import useFetchGifs from "../../hooks/useFetchGifs"
jest.mock("../../hooks/useFetchGifs");

describe("pruebas en gif grid", () => {

    const category = "One Punch"

    test("tomatchsnapshot", () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })


        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    })

    test("Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs", () => {

        const gifs = [{
            id:"ABC",
            url:"https://lochalhost/cualquiera.jpg",
            title:"MY title"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false)
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
        
    })
})