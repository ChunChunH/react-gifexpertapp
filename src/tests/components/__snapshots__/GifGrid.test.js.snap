// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gif grid Debe mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
   
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="MY title"
      url="https://lochalhost/cualquiera.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en gif grid tomatchsnapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
   
  <div
    className="card-grid"
  />
</Fragment>
`;
